tls-client/main.cpp

```diff
@@ -417,8 +417,11 @@ class HelloHTTPS {
  */
 int main() {
     mbedtls_platform_context platform_ctx;
-    if(mbedtls_platform_setup(&platform_ctx) != 0) {
-        return -1;
+    int ret = MBEDTLS_EXIT_FAILURE;
```

Comment:
I think this function will always return MBEDTLS_EXIT_FAILURE, a failure code...

Reply:
Ret is only used here to store the value of platform setup and print the error. When errors occur, sure, it returns MBEDTLS_EXIT_FAILURE, but when everything goes right, MBEDTLS_EXIT_SUCCESS is returned (at the end).